OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitWebViewFragment.java

```diff
@@ -102,17 +102,20 @@ public void onReceivedError(WebView view, int errorCode,
                     @TargetApi(Build.VERSION_CODES.M)
                     public void onReceivedHttpError(WebView view, WebResourceRequest request,
                                                     WebResourceResponse errorResponse) {
-                        didReceiveError = true;
-                        switch (errorResponse.getStatusCode()) {
-                            case HttpStatus.FORBIDDEN:
-                            case HttpStatus.UNAUTHORIZED:
-                            case HttpStatus.NOT_FOUND:
-                                EdxCookieManager.getSharedInstance(getContext())
-                                        .tryToRefreshSessionCookie();
-                                break;
+                        // If error occurred for web page request
+                        if (request.getUrl().toString().equals(view.getUrl())) {
```

Comment:
We can apply the following if check for the very safe side but we should generate some scenario first in which page could be loaded but is not getting load irrespective of some minor error otherwise it will be leap in the dark.

`if (failingUrl != null && failingUrl.equals(view.getUrl())) {`